compiler/src/iree/compiler/Codegen/Common/GPU/GPUPruneDPSOps.cpp

```diff
@@ -0,0 +1,129 @@
+// Copyright 2025 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/compiler/Codegen/Common/GPU/Passes.h"
+#include "iree/compiler/Codegen/Utils/GPUUtils.h"
+#include "iree/compiler/Dialect/TensorExt/IR/TensorExtOps.h"
+#include "llvm/ADT/STLExtras.h"
+#include "mlir/Dialect/Bufferization/IR/Bufferization.h"
+#include "mlir/Dialect/GPU/IR/GPUDialect.h"
+
+namespace mlir::iree_compiler {
+
+#define GEN_PASS_DEF_GPUPRUNEDPSOPSPASS
+#include "iree/compiler/Codegen/Common/GPU/Passes.h.inc"
+
+namespace {
+/// Pass to create a `bufferization.alloc_tensor` in private space for all DPS
+/// ops with unused results that can't be removed.
+/// These unused results, if originating from loads from global memory, trigger
+/// allocations in global memory space during bufferization, which will fail.
+/// So, the allocations must be made earlier to avoid failed bufferization.
+/// This pass is a hack to get around other work that can be done to improve
+/// the bufferization algorithm.
+/// For such cases, need to make Flow preserve the unused result as a
+/// result of the dispatch.
+struct GPUPruneDPSOpsPass final
```

Comment:
I think we need to rename the pass. There is no pruning, IMO. I'm not good at naming, but let me give it a try. How about `GPUAllocPrivateMemoryForDPSOpsPass`?

Also, please do not duplicate the comment. I think the documentation of the pass is in `Passes.td`. And the rest of comment can be:

```suggestion
/// This pass is a hack to get around other work that can be done to improve
/// the bufferization algorithm.
/// For such cases, need to make Flow preserve the unused result as a
/// result of the dispatch.
struct GPUPruneDPSOpsPass final
```